include/openssl/e_os2.h

```diff
@@ -206,8 +206,12 @@ extern "C" {
 #  endif
 # endif
 
-# if defined(OPENSSL_SYS_UEFI) && !defined(ssize_t)
-#  define ossl_ssize_t int
+# if defined(OPENSSL_SYS_UEFI) && !defined(ossl_ssize_t)
+#  if !defined(ssize_t)
+#   define ossl_ssize_t int
+#  else
+#   define ossl_ssize_t ssize_t
+#  endif
 #  define OSSL_SSIZE_MAX INT_MAX
```

Comment:
Thanks, dot-asm.
Yes, ssize_t will be defined in external UEFI header file, which is architecture-aware definition. So we check if the definition is ready, and then use the UEFI-version value inferred from the target. If not, 32-bit will be the default.
For OSSL_SIZE_MAX, we choose INT_MAX as the default value, since UEFI is the resource-restricted environment. The INT_MAX should be good enought for UEFI usage scenario.  This may looks a little inconsistency,  but it should have no side-effect. 
Or we can add SSIZE_MAX definition (arch-awared) in UEFI header file for consistency.